File: UIControlsDemo/app/src/main/java/com/example/uicontrolsdemo/MainActivity.java
package com.example.uicontrolsdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView textView;
    Button button1, button2;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);
        button1 = findViewById(R.id.button1);
        button2 = findViewById(R.id.button2);

        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textView.setText("左边");
            }
        });

        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textView.setText("右边");
            }
        });
    }




    @Override
    protected void onPause() {
        super.onPause();

        textView.setText("pause");
    }
}

File: VideoView/app/src/main/java/com/example/videoview/MainActivity.kt
package com.example.videoview

import android.media.PlaybackParams
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.MediaController
import androidx.lifecycle.lifecycleScope
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_main.view.*
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {

    private var savePosition = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        videoPlay2()

    }

    fun videoPlay0() {
        val videoPath = "android.resource://$packageName/${R.raw.testhtml5}"
        videoView.setVideoPath(videoPath)

        videoView.start()

    }

    fun videoPlay1() {
        val videoPath = "android.resource://$packageName/${R.raw.testhtml5}"
        videoView.setVideoPath(videoPath)
        videoView.setMediaController(MediaController(this))
        videoView.setOnPreparedListener {
//            it.seekTo()
            progressBar.max = it.duration
            it.isLooping = true
            it.playbackParams = PlaybackParams().apply {
//                speed = 2f
//                pitch = 2f
            }
            it.start()
        }
    }

    fun videoPlay2() {
//        val videoPath = "android.resource://$packageName/${R.raw.testhtml5}"
        val videoPath = "http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4"
        videoView.setVideoPath(videoPath)
        videoView.setOnPreparedListener {

            progressBar2.visibility = View.INVISIBLE
            progressBar.max = it.duration
            it.isLooping = true
            it.playbackParams = PlaybackParams().apply {
//                speed = 2f
//                pitch = 2f
            }
            it.seekTo(savePosition)
            it.start()
        }

        lifecycleScope.launch {
            while (true) {
                if (videoView.isPlaying) {
                    progressBar.progress = videoView.currentPosition
                }
                delay(500)
            }
        }
    }


    override fun onPause() {
        super.onPause()
        savePosition = videoView.currentPosition
    }
}